### ISSUE TITLE
Handle multiple devices (Proxyman instances)

### ISSUE THREAD

**Issue by djbe** [ProxymanApp/atlantis#72]
One thing we noticed during the implementation of [Atlantis for Android](#28), is that Atlantis only maintains one connection (task, see [here](https://github.com/ProxymanApp/atlantis/blob/main/Sources/Transporter.swift#L39)) in total.

Whenever it finds and resolves a new service (i.e. a new Proxyman instance), it connects to it. In the process of establishing that connection, it [closes any previous connections](https://github.com/ProxymanApp/atlantis/blob/main/Sources/Transporter.swift#L176).

It would be better if Atlantis would seamlessly handle multiple Proxyman instances, like Bagel does. It's extremely useful in an office environment, so we can easily help each other out in debugging network communication per device/app/...

**Comment by NghiaTranUIT** [comment]
Sorry for the late reply. I completely missed it.

At the moment, as you mention, Atlantis only maintains one connection. I don't think it's a good idea if we support multiple connections by default.

If one user starts Atlantis with default param:
```swift
Atlantis.start()
```
Then, all colleagues, who are opening Proxyman, can see the traffic. Even though they don't want to.

It also a reason why we introduce the hostname to connect to a specific computer.
```swift
Atlantis.start(hostName: "_your_host_name")
```

**Comment by djbe** [comment]
The thing is, Atlantis doesn't support multiple connections **at all** right now. What happens right now is more undefined behaviour, whichever is the latest Proxyman instance to be detected, is the one it will connect to (all the other ones it will connect to, and disconnect once a new one is discovered). 

Bagel has the same option of defining a hostname, and sure, keep that, it's good to have that. But Bagel by default supports multiple connections, which is much much easier to use for everyone.

But this does **not** work when you have multiple developers working on an app. Now everyone is changing the Atlantis startup code, to replace the host name with their own. You either have to remember to never commit that (which no one remembers), or people keep committing changes to that line (which dirties your git history with useless changes). Or even worse, sometimes because of network/system issues your hostname changes (to "devicename-1", "device-name-2", ...), and suddenly Atlantis stops working for all those users until they somehow find out that their hostname changed 🤷. 

So the **default** should be connect to all Proxyman instances. Easy to use for everyone, nothing they have to worry about. Only people that don't want connections from everyone should provide a hostname during startup.

**Comment by khoogheem** [comment]
Yep. I would love to see proxyman work correctly when you have the same Application running on Device and on Simulator.
Currently Proxyman shows both devices.. but the data that gets displayed seems to be the same across both devices, but it its the data from the first connected device

**Comment by NghiaTranUIT** [comment]
@RiyanFransen @djbe Good news that I'm working on this essential feature at https://github.com/ProxymanApp/atlantis/pull/90

I will test with multiple Proxyman app on different Mac devices this week and send you guys a beta build soon 🙌 

Thanks again for following up 🥇

**Comment by NghiaTranUIT** [comment]
and I would say sorry for the delay on this feature implementation 🙇

**Comment by djbe** [comment]
Fantastic news! :tada: Will have to try it when I'm back in the office (currently on holidays).

**Comment by NghiaTranUIT** [comment]
Feel free to keep it open until we're happy with this update 👍

**Comment by RiyanFransen** [comment]
@NghiaTranUIT Awesome big thanks! Unfortunately now we aren't permitted to go to the office with a big group so I won't be able to confirm the feature/fix in the foreseeable future. Feel free to close the issue, we'll give a heads up if we spot something!

**Comment by RiyanFransen** [comment]
@NghiaTranUIT We are happy with the update, issue can be closed. Many thanks for the enhancement!

**Comment by NghiaTranUIT** [comment]
You're welcome @RiyanFransen 🎉
